test/corpus_test.go

```diff
@@ -0,0 +1,128 @@
+package test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+	"text/template"
+)
+
+var (
+	TESTCASE_SEPERATOR      = regexp.MustCompile("(?m)^(=+)$")
+	INPUT_OUTPUT_SEPERATOR  = regexp.MustCompile("\n(---)\n")
+	TRIM_TESTCASE_SEPERATOR = regexp.MustCompile("^(=+)\n")
+	TESTS_DIR               = "./corpus"
+)
+
+func TestCorpus(t *testing.T) {
+	testCases, err := getTestCases()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Printf("Got %d test cases\n", len(testCases))
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			testTemplate(t, testCase.input, testCase.isError)
+		})
+	}
+}
+
+func testTemplate(t *testing.T, input string, isError bool) {
+	t.Helper()
+
+	input = fmt.Sprintf("{{ $var := 1}} \n %s", input) // prevent variable not found errors
+	_, err := template.New("template").Funcs(
+		template.FuncMap{
+			"pipeline":     func() string { return "" },
+			"functionName": func() string { return "" },
+			"condition":    func() string { return "" },
+		},
+	).Parse(input)
+
+	if err != nil && !isError {
+		t.Fatal(err)
+	} else if err == nil && isError {
+		t.Fatal("expected error")
+	}
+}
+
+type TestCase struct {
+	file        string
+	name        string
+	input       string
+	sExpression string
+	isError     bool
+}
+
+func NewTestCase(file, name, input, sExpression string) TestCase {
+	return TestCase{
+		name:        fmt.Sprintf("%s: %s", file, name),
+		input:       input,
+		sExpression: sExpression,
+		isError:     strings.Contains(sExpression, "(ERROR") || strings.Contains(sExpression, "(MISSING"),
+	}
+}
+
+func getTestCasesForFile(filename string, content string) []TestCase {
+	testCases := []TestCase{}
+	parts := []string{}
+
+	// split and remove empty parts
+	for _, part := range TESTCASE_SEPERATOR.Split(content, -1) {
+		if part != "" {
+			parts = append(parts, part)
+		}
+	}
+
+	for i := 0; i < len(parts)-1; i += 2 {
+		testName := strings.TrimSpace(parts[i])
+		testContent := parts[i+1]
+		testParts := INPUT_OUTPUT_SEPERATOR.Split(testContent, -1)
+
+		if len(testParts) != 2 {
+			fmt.Printf("Error parsing %s: Testcase has invalid format %s\n", filename, testName)
+			continue
```

Comment:
Maybe we should just set testParts to something like this:
```suggestion
			testParts = []string{"", ""}
```
 and check in the `testTemplate` against `input` so we get a test failure from a bad formatted test.

Reply:
With this change, I can see a test failure easily:
`    --- FAIL: TestCorpus/corpus/actions.txt:_invalid_comments (0.00s)`
(although this stops failing once the regex change suggested above is applied).